### PR TITLE
fix!: Make Error type less brittle

### PR DESCRIPTION
**Pull request by epage**
This is part of #11

BREAKING CHANGE: `Error::kind` member is no longer available.  Instead
use `Error::kind()`.  `ErrorKind` can no longer be exhaustively matched.

### CODE REVIEW

**Comment by JeanMertz** [comment]
This looks great, thanks!

By the way, as an aside, I came across this project which uses Conventional Commits, but in order for contributors not to have to worry about that too much, they simply squash all PRs into a single commit, and require the PR title to be a correct semantic commit message:

https://github.com/timberio/vector

Check out their [CONTRIBUTING.md](https://github.com/timberio/vector/blob/3ce0b4ed645d2844f1f6c5308409e2e9466c0799/CONTRIBUTING.md#github-pull-requests) document.

It's something that I have been thinking about as well, as it does increase the barrier to entry when you require individual commits in a PR to be correctly formatted, or it requires more maintenance by the repo owner by rewriting commits in PRs (and rewriting commit messages of others in a PR might even come across as a bit pedantic/annoying to the contributors themselves).

I haven't played with their strategy yet, but it seems to work well for them.

The only downside I noticed is that if you use GitHub's "squash and merge", you'll get commit messages like these:

https://github.com/timberio/vector/commit/fc2c1db5824f8499190efa078c993f3f52737043

Which, as you can see, results in footers that are kind of awkward, because this:

```
Signed-off-by: Jean Mertz <xxx>

* refactor: only call Captures::get if required

Signed-off-by: Jean Mertz <xxx>

* fixup! feat: support default variable values

Signed-off-by: Jean Mertz <xxx>

* fixup! feat: support default variable values

Signed-off-by: Jean Mertz <xxx>
```

Basically turns into 4 footers, with the top three footers also containing the next line after `Signed-off-by`, which is actually a commit message of the squashed commits.

**Comment by epage** [comment]
> By the way, as an aside, I came across this project which uses Conventional Commits, but in order for contributors not to have to worry about that too much, they simply squash all PRs into a single commit, and require the PR title to be a correct semantic commit message:

The big question for this is how much someone puts into a PR.  Some people go to one extreme of a PR being about 10 lines where squashing doesn't work well.  Others effectively create long-lived branches which you'd never squash.  For me, I'm somewhere in between.  Sometimes you don't know what refactors you might need until you are implementing the feature.  I then split these up into multiple commits and have them in a single PR (I describe PRs as "telling a story").  That is not something that should be squashed.  In contrast, if the person is puts all of their sloppy work up and is unfamiliar with [how to clean it up](https://epage.github.io/dev/commits/), then it can make sense to squash it for them. 

In my projects, I normally give people a choice, leaving it to their judgement of what they feel comfortable with on whether I squash or not as long as their change is small enough.  I do give them the option to be mentored on it.  If its big, I'll tell them to split it and it is up to them to figure it all out.
